pkg/controller/controller.go

```diff
@@ -787,6 +791,11 @@ func getProvisionSecretReference(secretParams secretParamsMap, storageClassParam
 // - the resolved name is not a valid secret name
 // - the resolved namespace is not a valid namespace name
 func getSecretReference(secretParams secretParamsMap, storageClassParams map[string]string, pvName string, pvc *v1.PersistentVolumeClaim) (*v1.SecretReference, error) {
+	// Ensure that no unsupported SC parameters are passed in.
```

Comment:
This will cause secret templates for other operations to fail. It's probably simplest to just let resolveTemplate() fail.

Reply:
Makes sense. Removed this validation.